airflow/providers_manager.py

```diff
@@ -882,7 +884,11 @@ def _add_widgets(self, package_name: str, hook_class: type, widgets: dict[str, A
                 # In case of inherited hooks this might be happening several times
                 continue
             self._connection_form_widgets[prefixed_field_name] = ConnectionFormWidgetInfo(
-                hook_class.__name__, package_name, field, field_identifier
+                hook_class.__name__,
+                package_name,
+                field,
+                field_identifier,
+                issubclass(field.field_class, PasswordField),
```

Comment:
I wonder if `field.field_class.widget.input_type == "password"` would be better. It covers more custom fields as well.

Reply:
Yeah. You are right. I looked at how PasswordField is defined, and it's "passwordness" is determined by the widget, so it will be better with the wideget check - but not evrey widget has input_type, so slightly different check is better.